File: 0088-merge-sorted-array/0088-merge-sorted-array.swift
class Solution {
    func merge(_ nums1: inout [Int], _ m: Int, _ nums2: [Int], _ n: Int) {
        var idx1 = m
        var idx2 = n

        while idx1 > 0 || idx2 > 0 {
            if idx2 == 0 {
                nums1[idx1 + idx2 - 1] = nums1[idx1 - 1]
                idx1 -= 1
            } else if idx1 == 0 {
                nums1[idx1 + idx2 - 1] = nums2[idx2 - 1]
                idx2 -= 1
            } else if nums1[idx1 - 1] >= nums2[idx2 - 1] {
                nums1[idx1 + idx2 - 1] = nums1[idx1 - 1]
                idx1 -= 1
            } else {
                nums1[idx1 + idx2 - 1] = nums2[idx2 - 1]
                idx2 -= 1
            }
        }

    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        while m + n > 0:
            if m == 0:
                nums1[m + n - 1] = nums2[n - 1]
                n -= 1
            elif n == 0:
                nums1[m + n - 1] = nums1[m - 1]
                m -= 1
            elif nums1[m - 1] >= nums2[n - 1]:
                nums1[m + n - 1] = nums1[m - 1]
                m -= 1
            else:
                nums1[m + n - 1] = nums2[n - 1]
                n -= 1